bedrock/contentful/utils.py

```diff
@@ -0,0 +1,68 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+from collections import defaultdict
+from typing import List
+
+from django.conf import settings
+
+from bedrock.contentful.models import ContentfulEntry
+
+
+def can_use_locale(
+    content_type: str,
+    classification: str,
+    locale: str,
+    default_locale: str = "en-US",
+) -> bool:
+    """Is there enough of this content type, of this classification,
+    in the given locale, compared to the default locale to, for example,
+    show a listing page of it?"""
+
+    active_locales = get_active_locales(
+        content_type,
+        classification,
+        default_locale,
+    )
+    return locale in active_locales
+
+
+# TODO? Cache this
+def get_active_locales(
+    content_type: str,
+    classification: str,
+    default_locale: str = "en-US",
+    slug: str = None,
+) -> List:
+    """How many locales do we have 'enough' of to consider them an active locale?"""
+
+    kwargs = dict(
+        content_type=content_type,
+        classification=classification,
+    )
+    if slug:
```

Comment:
Just making sure I'm following correctly. This is used for the main view of the resource center when called without a slug. Then when it is called with a slug, the percentages below should all be 100 since each active locale should have 1 article? Is that right?

Reply:
Looking back at this now, I think this can be made easier with an early return if we have a slug - once we call
```
all_locales = ContentfulEntry.objects.filter(**kwargs).values_list("locale", flat=True)
```
then that's all the locales for that slug + content type + classification so they are all valid. There's no need to do the extra legwork of the threshold check at all.

Will update things now

Reply:
(This is what comes from overloading something build for A to do A+B without stepping back to see the bigger picture 🤦 )